integration/spark-common/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchemaCommon.scala

```diff
@@ -378,19 +376,65 @@ class AlterTableColumnSchemaGenerator(
       }
     }
 
+    var isLocalDictEnabledForMainTable = tableSchema.getTableProperties
+      .get(CarbonCommonConstants.LOCAL_DICTIONARY_ENABLE)
+    if (null == isLocalDictEnabledForMainTable) {
+      isLocalDictEnabledForMainTable = CarbonProperties.getInstance()
+        .getProperty(CarbonCommonConstants.LOCAL_DICTIONARY_SYSTEM_ENABLE,
+          CarbonCommonConstants.LOCAL_DICTIONARY_SYSTEM_ENABLE_DEFAULT)
+    }
+
+    val alterMutableTblProperties: scala.collection.mutable.Map[String, String] = mutable
+      .Map(alterTableModel.tableProperties.toSeq: _*)
```

Comment:
Please check do we need null check for alterTableModel.tableProperties

Reply:
no need for null check, beacuse we are always adding sort columns